src/resolution.ts

```diff
@@ -16,7 +16,7 @@ import { flatMapExpression } from "./dsl";
 /**
  * TypeRefResolution is the result of resolving a type reference.
  */
-export interface TypeRefResolution {
+export type TypeRefResolution = {
```

Comment:
This is from turning on the eslint rule that prefers `type` declarations to `interface`s. They're mostly equivalent, but you can use `type` everywhere so I'd rather use it.